packages/inferno/src/DOM/hydration.ts

```diff
@@ -128,17 +158,33 @@ function hydrateElement(vNode: VNode, dom: Element, lifecycle: LifecycleClass, c
 	return dom;
 }
 
-function hydrateChildren(children: InfernoChildren, parentDom: Element, lifecycle: LifecycleClass, context: Object, isSVG: boolean): void {
+function hydrateChildren(
+	children: InfernoChildren,
+	parentDom: Element,
+	lifecycle: LifecycleClass,
+	context: Object,
+	isSVG: boolean
+): void {
 	normalizeChildNodes(parentDom);
 	let dom = parentDom.firstChild;
 
 	if (isArray(children)) {
-		for (let i = 0, len = (children as Array<string | number | VNode>).length; i < len; i++) {
-			const child = children[ i ];
+		for (
```

Comment:
@vjeux

- length on own line changes scope of the variable

-  it isn't a cast <img width="1257" alt="screen shot 2017-05-07 at 12 54 53 am" src="https://cloud.githubusercontent.com/assets/4022631/25778872/ce89c46c-32bf-11e7-826b-05f64042e8e8.png">

- for of can be a lot slower https://jsperf.com/foreach-vs-for-loop-vs-for-in-vs-for-of-vs-babel-for-of